File: 633-sum-of-square-numbers/sum-of-square-numbers.java
class Solution {
    public boolean judgeSquareSum(int c) {
        long l = 0,r=(long)Math.sqrt(c);
        while(l<=r){
            long one = l*l;
            long two = r*r;
            long sum = one+two;
            if(sum==c) return true;
            else if(sum<c) l++;
            else r--;
        }
        return false;
    }
}

File: 633-sum-of-square-numbers/sum-of-square-numbers.py
class Solution(object):
    def judgeSquareSum(self, c):
        l=0
        r=int(math.sqrt(c))
        print(r)
        while l<=r:
            one = l*l
            two = r*r
            sum = one+two
            if sum == c:
                return True
            elif sum<c:
                l+=1
            else:
                r-=1
        return False
        